roles/kubernetes/preinstall/tasks/0020-set_facts.yml

```diff
@@ -14,34 +7,6 @@
   tags:
     - facts
 
-- name: Set os_family fact for other debian-based operating systems
-  set_fact:
-    ansible_os_family: "Debian"
-  when: ansible_distribution in debian_os_family_extensions
```

Comment:
The `debian_os_family_extensions` should not be deleted. For Linux distributions not currently supported by Ansible, they can be added to this parameter to determine the specific `os_family` type.

Reply:
below (mail can't answer to threads) + see commit message of df30c01ce

Reply:
Could we keep `debian_os_family_extensions` as an empty array to allow for future expansion of new unknown distributions?

Reply:
I really don't want to keep dead code around "just in case". -> YAGNI.
Besides, if we really need to, we can always bring it back.

Reply:
I believe it is necessary. Please keep it. Thank you.

Reply:
Every piece of code is a future mental load on every future reader of the code.
If that piece of code never run, this is just dead weight. We already have a lot of that in kubespray, and we *need* to keep it as low as possible.

Furthermore, it is not at all guaranteed that it keeps working enough for it to be useful in a hypothetical future where we would need it again, while the rest of the code changes.

If you think there is concrete benefits to keeping it, please state them. Otherwise, let's get rid of it.

Reply:
Also now that we actually have an up to date Ansible version, it's probably likely that Ansible would get the support quicker than us anyway and we would have to continue regularly update to get the new distribution "for free".

Reply:
Although Ansible already supports Uos, the reality is that Uos has [multiple categories of versions](https://doc.chinauos.com/content/jc2clIgBZybsW1rIbBg2). 
Some are based on Debian, while others are based on Red Hat. 
This may sound strange, but it is the actual situation. 
The `ansible_distribution` value obtained from different versions of Uos can vary. 
Here are some of the common values: `UniontechOS`, `UnionTech`, `UnionTech OS Server 20`. 
There may be others, and some versions of `ansible_distribution` may contain special characters, making the identification of Uos difficult. 
The pull request addressed in https://github.com/ansible/ansible/pull/77275 resolves only one of these scenarios.

Furthermore, the feature `debian_os_family_extensions` has been [utilized in the Daocloud DCE 5.0 product](https://docs.daocloud.io/en/install/os-install/otherlinux/#__tabbed_1_1). 
Many customers were able to deploy clusters in different Uos versions because of this feature. 
Therefore, I believe that until these issues in Uos are completely resolved, it is preferable to put this breaking change on hold.

@VannTen @MrFreezeex @yankay @mzaian @floryut @cyclinder @ChanthMiao

Reply:
> Although Ansible already supports Uos, the reality is that Uos has [multiple categories of versions](https://doc.chinauos.com/content/jc2clIgBZybsW1rIbBg2).

Unfortunately I can't read this :( 

>  Some are based on Debian, while others are based on Red Hat.
> ...

Yeah, I understand that, as pointed out in the other thread.

> Furthermore, the feature `debian_os_family_extensions` has been [utilized in the Daocloud DCE 5.0 product](https://docs.daocloud.io/en/install/os-install/otherlinux/#__tabbed_1_1).

Ok, so this is your point, that this variable is in fact an used from the inventory. I see. It's unfortunate that this got in defaults instead of vars, but I see what you mean.

So going from there:
- eventually, we should get rid of this, can we agree on that ? This has no place in Kubespray and should be in Ansible, and should be upstreamed in Ansible.
- an alternative solution (in fact, a better solution IMO) is to set directly `ansible_os_family` in group_vars for the Uos OS when deploying. -> I'd even be willing to put that in #10983 , in a Uos vars file. (we can extend the first_found logic to include other keys of os_release, if needed).

The first of these items can take time, but it should absolutely happens at some point. The second is a workaround which does not impose the load on the kubespray project.

Reply:
> eventually, we should get rid of this, can we agree on that ? This has no place in Kubespray and should be in Ansible, and should be upstreamed in Ansible.

We are in agreement on this point.

> an alternative solution (in fact, a better solution IMO) is to set directly ansible_os_family in group_vars for the Uos OS when deploying. -> I'd even be willing to put that in https://github.com/kubernetes-sigs/kubespray/pull/10983 , in a Uos vars file. (we can extend the first_found logic to include other keys of os_release, if needed).

It sounds feasible. I will verify it later. Thank you for your suggestion.

Reply:
/lgtm